udev_device.c

```diff
@@ -540,6 +559,70 @@ static void set_properties_from_props(struct udev_device *udev_device)
     udev_list_entry_add(&udev_device->properties, "ID_PATH", id, 0);
 }
 
+static int set_properties_from_helper(struct udev_device *udev_device)
+{
+    char *envp[UEVENT_NUM_ENVP];
+    char env[UEVENT_BUFFER_SIZE];
+
+    char *ptr = env;
+    size_t i = 0;
+    struct udev_list_entry *entry;
+
+    for (entry = udev_device_get_properties_list_entry(udev_device);
+         entry;
+         entry = udev_list_entry_get_next(entry)) {
+        const char *name = entry->name;
+        const char *value = entry->value ? entry->value : "";
+
+        // '{name}={value}\0'
+        size_t max_size = strlen(name) + 1 + strlen(value) + 1;
+        snprintf(ptr, max_size, "%s=%s", name, value);
+
+        envp[i] = ptr;
+        ptr += max_size;
+        i++;
+    }
+
+    envp[i] = NULL;
+
+    int out_pipe[2];
+
+    if (pipe(out_pipe)) {
+        return -1;
+    }
+
+    posix_spawn_file_actions_t actions;
+
+    posix_spawn_file_actions_init(&actions);
+    posix_spawn_file_actions_addclose(&actions, out_pipe[0]);
+    posix_spawn_file_actions_adddup2(&actions, out_pipe[1], 1);
+    posix_spawn_file_actions_addclose(&actions, out_pipe[1]);
```

Comment:
I wonder if we should no-op(or close?) stdin/stderr using /dev/null to prevent child process from inheriting parent's stdin/stderr